src/network/network_lan_promotion.cc

```diff
@@ -98,8 +98,8 @@ LanBase::LanBase(uint16_t port) : socket_v4(io_service), socket_v6(io_service) {
 		if (ifa->ifa_addr == nullptr) {
 			continue;
 		}
-		if (!(ifa->ifa_flags & IFF_LOOPBACK) && !(ifa->ifa_flags & IFF_BROADCAST) &&
-		    !(ifa->ifa_flags & IFF_MULTICAST)) {
+		if (((ifa->ifa_flags & IFF_LOOPBACK) == 0u) && ((ifa->ifa_flags & IFF_BROADCAST) == 0u) &&
```

Comment:
just for my information would it be possiblke to just sequentially do bitwise and with all 3 flags and comnpare only once if it is 0u
like (ifa->ifa_flags & IFF_LOOPBACK & IFF_BROADCAST & IFF_MULTICAST) == 0u

Reply:
No, bitwise `AND` sets all bits to 0 except the ones which are present in both variables. So the expression `(flags & MASK)` always evaluates to either 0 or `MASK`. In both cases the expression `(flags & MASK) & OTHER_MASK` would be 0 so this is not equivalent.

What we could here do is to combine the three masks with bitwise `OR` and check if that mask matches anything:
```c++
if (a->ifa_flags & (IFF_LOOPBACK | IFF_BROADCAST | IFF_MULTICAST) == 0u)
```
but this only works in some cases. 
However, bitwise m~agic~ath is always hard to read, so we should IMHO use the same syntax everywhere (namely, only one mask per expression) to make it a bit easier to understand and therefore less bug-prone.